crates/sc-subspace-block-relay/src/utils.rs

```diff
@@ -90,3 +96,68 @@ impl NetworkPeerHandle {
             .map_err(|err| RequestResponseErr::DecodeFailed { response_len, err })
     }
 }
+
+/// Convenience wrapper around prometheus counter, which can be optional.
+pub(crate) struct RelayCounter(Option<Counter<U64>>);
+
+impl RelayCounter {
+    /// Creates the counter.
+    pub(crate) fn new(
+        name: &str,
+        help: &str,
+        registry: Option<&Registry>,
+    ) -> Result<Self, PrometheusError> {
+        let counter = if let Some(registry) = registry {
+            Some(register(Counter::new(name, help)?, registry)?)
+        } else {
+            None
+        };
```

Comment:
This feels like the wrong level of abstraction to have `Option`, is this how it was done in Substrate? I think we should always have registry (the official one, not from Susbtrate), it'll just not always expose HTTP endpoint for metrics.

Reply:
This is based on receiving the registry from config (same way DSN was doing it), which appears to be optional. These are just convenience wrappers for now, as these conditional checks all over the code was getting ugly.

When the registry becomes non-optional, we should be able to just search replay these RelayCounter* with the actual prometheus strucs.